source/extensions/filters/http/basic_auth/basic_auth_filter.cc

```diff
@@ -97,8 +100,13 @@ bool BasicAuthFilter::validateUser(const UserMap& users, absl::string_view usern
 Http::FilterHeadersStatus BasicAuthFilter::onDenied(absl::string_view body,
                                                     absl::string_view response_code_details) {
   config_->stats().denied_.inc();
-  decoder_callbacks_->sendLocalReply(Http::Code::Unauthorized, body, nullptr, absl::nullopt,
-                                     response_code_details);
+  decoder_callbacks_->sendLocalReply(
+      Http::Code::Unauthorized, body,
+      [uri = this->original_uri_](Http::ResponseHeaderMap& headers) {
+        std::string value = absl::StrCat("Basic realm=\"", uri, "\"");
+        headers.setCopy(Http::Headers::get().WWWAuthenticate, value);
+      },
+      absl::nullopt, response_code_details);
```

Comment:
Thanks for the contribution. LGTM overall.

But I don't think we need to create and store the `original_uri_` as a member of filter. You could simply capture `this` in the lambda and build the value in the lambda.

Tips: You can get request headers by the `decoder_callbacks_->requestHeaders()` in the lambda.